test/SignalRServiceExtension.Tests/Trigger/SignalRTriggerTests.cs

```diff
@@ -50,8 +48,72 @@ public async Task CreateListenerTest()
+            var context = new InvocationContext { Arguments = new object[] { Guid.NewGuid().ToString(), Guid.NewGuid().ToString() } };
+            var triggerContext = new SignalRTriggerEvent { Context = context };
+            var result = await binding.BindAsync(triggerContext, null);
+            var bindingData = result.BindingData;
+            Assert.Equal(context.Arguments[0], bindingData["arg0"]);
+            Assert.Equal(typeof(string), binding.BindingDataContract["arg0"]);
+            Assert.Equal(context.Arguments[1], bindingData["arg1"]);
+            Assert.Equal(typeof(string), binding.BindingDataContract["arg1"]);
+        }
+
+        [Fact]
+        public async Task BindingDataTestWithMoreParameterNames()
+        {
+            var binding = CreateBinding(nameof(TestFunctionWithTwoStringArgument), new string[] { "arg0", "arg1", "arg2" });
+            var context = new InvocationContext { Arguments = new object[] { Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString() } };
+            var triggerContext = new SignalRTriggerEvent { Context = context };
+            var result = await binding.BindAsync(triggerContext, null);
+            var bindingData = result.BindingData;
+            Assert.Equal(context.Arguments[0], bindingData["arg0"]);
+            Assert.Equal(typeof(string), binding.BindingDataContract["arg0"]);
+            Assert.Equal(context.Arguments[1], bindingData["arg1"]);
+            Assert.Equal(typeof(string), binding.BindingDataContract["arg1"]);
+            Assert.Equal(context.Arguments[2], bindingData["arg2"]);
+            Assert.Equal(typeof(object), binding.BindingDataContract["arg2"]);
+        }
+
+        [Fact]
+        public async Task BindingDataTestWithUnmatchedParameterNamesAndInvocation()
+        {
+            var binding = CreateBinding(nameof(TestFunctionWithTwoStringArgument), new string[] { "arg0", "arg1", "arg2" });
+            // Less invocation arguments than ParameterNames
+            var context = new InvocationContext { Arguments = new object[] { Guid.NewGuid().ToString(), Guid.NewGuid().ToString() } };
+            var triggerContext = new SignalRTriggerEvent { Context = context };
+            await Assert.ThrowsAsync<SignalRTriggerParametersNotMatchException>(() => binding.BindAsync(triggerContext, null));
+        }
+
+        private SignalRTriggerBinding CreateBinding(string functionName, string[] parameterNames)
+        {
+            var parameterInfo = this.GetType().GetMethod(functionName).GetParameters()[0];
+            var dispatcher = new TestTriggerDispatcher();
+            return new SignalRTriggerBinding(parameterInfo, new SignalRTriggerAttribute{ParameterNames = parameterNames}, dispatcher);
+        }
+
         public void TestFunction(InvocationContext context)
         {
         }
+
+        public void TestFunctionWithTwoStringArgument(InvocationContext context, string arg0, string arg1)
```

Comment:
can we use internal method to avoid add `SuppressMessage`?

Reply:
How about [ConnectionStringSettingFacts](https://github.com/Azure/azure-functions-signalrservice-extension/blob/dev/test/SignalRServiceExtension.Tests/JobhostEndToEnd.cs#L101), you don't need to add `SuppressMessage`

Reply:
I tried internal, and it works and easy.